README.md

```diff
@@ -1,96 +1,119 @@
+      --commit-email string      the email to use for commits when creating branches
+      --commit-message string    the msg to use on the resultant commit and pull request
+      --commit-name string       the name to use for commits when creating branches
+      --debug                    additional debug logging output
+      --from string              source Git repository
+  -h, --help                     help for promote
+      --insecure-skip-verify     Insecure skip verify TLS certificate
+      --keep-cache               whether to retain the locally cloned repositories in the cache directory
+      --repository-type string   the type of repository: github, gitlab or ghe (default "github")
+      --service string           service name to promote
+      --to string                destination Git repository
+
+Global Flags:
+      --github-token string   oauth access token to authenticate the request
 ```
 
-## Getting started
-
-At a high level the services command currently:
-
-- git clones the source and target repositories into ~/.promotion/cache
-- creates a branch
-- checks out the branch
-- copies the relevant files from the cloned source into the cloned target
-- pushes the cloned target
-- creates a PR from the new branch in the target to master in the target
+This will _copy_ all files under `/services/service-a/base/config/*` in `first-environment` to `second-environment`, commit and push, and open a PR for the change. Any of these arguments may be provided as environment variables, using all upper case and replacing `-` with `_`. Hence you can set CACHE_DIR, COMMIT_EMAIL, etc.
 
-See the [tekton-example](./tekton-example/README.md) directory for more on using the `promote` task with Tekton. See [automerge](./automerge/README.md) for some suggestions as to how promotion PullRequests may be merged automatically where appropriate.
+- `--cache-dir` : path on the local filesystem in which Git checkouts will be cached.
+- `--commit-email` : Git commits require an associated email address and username. This is the email address. May be set via ~/.gitconfig.
+- `--commit-message` : use this to override the commit message which will otherwise be generated automatically.
+- `--commit-name` : The other half of `commit-email`. Both must be set.
+- `--debug` : prints extra debug output if true.
+- `--from` : an https URL to a GitOps repository for 'remote' cases, or a path to a Git clone of a microservice for 'local' cases.
+- `--help`: prints the above text if true.
+- `--insecure-skip-verify` : skip TLS cerificate verification if true. Do not set this to true unless you know what you are doing.
+- `--keep-cache` : `cache-dir` is deleted unless this is set to true. Keeping the cache will often cause further promotion attempts to fail. This flag is mostly used along with `--debug` when investigating failure cases. 
+- `--repository-type` : the type of repository: github, gitlab or ghe (default "github")
```

Comment:
How do you provide the URL for accessing the `ghe` installation?

Reply:
I've pushed a small addition to the `--repository-type` argument. The API endpoint is worked out for you in https://github.com/rhd-gitops-example/services/blob/d3bfd0f0701791aa689ca08dac46285708b24216/pkg/git/client.go#L19 - so `--from` and `--to` need to be of the same type.